_posts/2016-9-5-Bonsai-Water-2.md

```diff
@@ -0,0 +1,176 @@
+
+
+For the Echo Blinds project, we wrote a whole <a href="https://grails.org/" target="\_blank">Grails application</a> for the relay server. We used an embedded database to store the objects, used Grails controllers to create the REST endpoints, and Grails views to make an API tester. This approach worked (and can be found <a href="https://github.com/TheConnMan/Message-Relay" target="\_blank">here</a> if you're interested), but it was just too heavy. There's no need for a full application server and a <a href="https://jenkins.io/" target="\_blank">Jenkins</a> instance to build the project and 750MB of RAM just to sit around for 99.9% of the day. The bottom line is that we're cheap, we didn't want to pay for that instance, so we built the Slack relay as a super simple Node.js projects.
+
+So without further ado, let's go through the steps!
+
+<div class="numbers">
+	<b>1</b>
+</div>
+
+### Setup AWS.
+
+We're going to need some setup before we can run **Slack Relay**, because **Slack Relay** runs best in a Docker container. AWS has a nice free tier you can use to run a small VM for free for a year. You can setup an AWS account and an EC2 region by following <a href="http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/get-set-up-for-amazon-ec2.html" target="\_blank">this guide</a>. A few notes for when you setup are:
+
+- You will need a key pair to connect to your instance. The key pair will be generated when you set up EC2.
+- Your security group needs to be open on port 80 to your home IP (it can be open to the world for testing, but it is recommended to only open it to your IP so no one else can access it)
+
+Once you have completed the above guide go to <a href="http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/EC2_GetStarted.html" target="\_blank">this guide</a> to set up your EC2 instance. t2.micro instance types can be used for free and are large enough for running our Docker container. When selecting an AMI for your VM choose **Ubuntu 14.04**.
+
+<div class="numbers">
+	<b>2</b>
+</div>
+
+### Setup Docker.
+Now that you have an EC2 instance, we'll need to connect to it to set up Docker. <a href="http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/AccessingInstancesLinux.html" target="\_blank">SSH into your EC2 instance</a>. For Ubuntu you can connect by running `ssh -i /path/to/mykey.pem ubuntu@ec2-my-ip.compute-1.amazonaws.com` in your favorite terminal application. Once you have connected, run the following to install Docker: ```bash sudo su curl -sSL https://get.docker.com/ | sh```
+
+And that's it, you've got Docker installed. To verify run `docker run hello-world`. You'll see the `hello-world` image get pulled down and echo a little "Hello World" message to you. Now we're ready to dive into **Slack Relay**.
```

Comment:
We're not really diving into **Slack Relay** after this so the last sentence doesn't make much sense.